client/src/dialogs/feedback/WeaknessesDialog.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import Grid from "@material-ui/core/Grid";
+import Dialog from "@material-ui/core/Dialog";
+
+import { useLocation } from "react-router-dom";
+import {
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  QuestionTextAndNo,
+  FeedbackBlueButton,
+  FeedbackOutlinedButton,
+  DialogCustomTextField,
+} from "../../components/DialogCommonComponents";
+
+const useStyles = makeStyles((theme) => ({
+  dialogActions: {
+    paddingTop: theme.spacing(4),
+    paddingBottom: theme.spacing(8),
+  },
+}));
+
+const WeaknessesDialog = ({
+  onClose,
+  onPreviousQuestionClick,
+  onNextQuestionClick,
+}) => {
+  const classes = useStyles();
+  const location = useLocation();
+
+  let openDialog = false;
+  const { pathname } = location;
+
+  if (
+    pathname === "/dashboard/feedback/4" ||
+    pathname === "/dashboard/feedback/4/"
+  )
```

Comment:
is this really necessary?

Reply:
On second thoughts not  necessary. I will just keep `if (pathname === "/dashboard/feedback/4/") {} `

Reply:
Okay so it looks like you are handling the dialogs path using this if statement. I have a suggestion below on how you can handle this differently